test/deploy/rackhd_ha_resource_install.py

```diff
@@ -0,0 +1,112 @@
+        # check interface for virtual ip
+        if "172.31.128" not in ip:
+            nic = "ens160"
+        else:
+            nic = "ens192"
+        command = ("crm configure primitive {} ocf:heartbeat:IPaddr2 " +
+                   "params ip='{}' nic='{}' " +
+                   "op monitor interval='10s' meta is-managed='true'").format(rsc_ip, ip, nic)
+        rc = fit_common.remote_shell(command, vmnum=vmnum)['exitcode']
+        return rc == 0
+
+    def get_southbound_network(self):
+        endpoints = fit_common.fitrackhd()['httpEndpoints']
+        southbound = self.find_southbound(endpoints)
+        if southbound and 'address' in southbound:
+            address = southbound["address"]
+            addrsplit = address.split('.')
+            return("{}.{}.{}".format(addrsplit[0], addrsplit[1], addrsplit[2]))
+        return None
+
+    def find_southbound(self, httpEndpoints):
+        for i in httpEndpoints:
+            if i["routers"] == "southbound-api-router":
+                return i
+        return None
+
+    def create_mongo_config(self, ip_list):
+        template_folder = './config_templates'
+        env = Environment(loader=FileSystemLoader(template_folder))
+        template = env.get_template("mongo_init.py")
+        config = {'_id': 'mongo_rs', 'version': 1, 'members': []}
+        for idx, val in enumerate(ip_list):
+            config['members'].append({'_id': idx, 'host': '{}:27017'.format(val)})
+        rendered = template.render(mongo_list=config)
+        return rendered
+
+    def test01_install_mongo_resource(self):
+        # create resource on first active node
+        vmnum = nodelist[0]
+        for mongo in range(1, numrs + 1):
+            # start mongo container as pacemaker resource
+            rsc = 'docker_mongo_{}'.format(mongo)
+            command = ("crm configure primitive {} ocf:heartbeat:docker " +
+                       "params allow_pull=true image='registry.hwimo.lab.emc.com/mongo' " +
+                       "run_opts=\\\'--privileged=true --net='host' -d -p 27017:27017\\\' " +
+                       "run_cmd=\\\'--replSet mongo_rs --logpath /var/log/mongodb/mongod.log\\\' " +
+                       "meta is-managed='true'").format(rsc)
+            self.assertEqual(fit_common.remote_shell(command, vmnum=vmnum)['exitcode'], 0, "{} resource failure.".format(rsc))
+            # create mongo virtual ip resource
+            sb_net = self.get_southbound_network()
+            if sb_net:
+                ip = '{}.12{}'.format(sb_net, mongo)
+                vip_dict['mongo'].append(ip)
+                rsc_ip = 'mongo_addr_{}'.format(mongo)
+                self.assertTrue(self.configure_virtual_ip_resource(vmnum, ip, rsc_ip), "{} resource failure.".format(rsc_ip))
+            # colocate mongo and virtual IPs
+            mongo_cls = 'mongo{}'.format(mongo)
+            command = "crm configure colocation {} inf: {} {}".format(mongo_cls, rsc, rsc_ip)
```

Comment:
Not sure how hound is missing this but I'm pretty sure rsc_ip will be undefined if sb_net is None.